tests/docker/docker_test.go

```diff
@@ -0,0 +1,207 @@
+
+	tc = projects.NewTestContext()
+
+	projectName := fmt.Sprintf("%s_%s", "TestLogInfoMetric", time.Now())
+
+	tc.InitProject(projectName)
+
+	for _, node := range tc.GetNodeDescriptions() {
+		edgeNode := tc.GetController().GetEdgeNode(node.Name)
+		if edgeNode == nil {
+			edgeNode = tc.NewEdgeNode(tc.WithNodeDescription(node), tc.WithCurrentProject())
+		} else {
+			tc.UpdateEdgeNode(edgeNode, tc.WithCurrentProject(), tc.WithDeviceModel(node.Model))
+		}
+
+		tc.ConfigSync(edgeNode)
+
+		if edgeNode.GetState() == device.NotOnboarded {
+			log.Fatal("Node is not onboarded now")
+		}
+
+		tc.AddNode(edgeNode)
+	}
+
+	tc.StartTrackingState(false)
+
+	res := m.Run()
+
+	os.Exit(res)
+}
+
+//checkAppDeployStarted wait for info of ZInfoApp type with mention of deployed AppName
+func checkAppDeployStarted(appName string) projects.ProcInfoFunc {
+	return func(msg *info.ZInfoMsg) error {
+		if msg.Ztype == info.ZInfoTypes_ZiApp {
+			if msg.GetAinfo().AppName == appName {
+				return fmt.Errorf("app found with name %s", appName)
+			}
+		}
+		return nil
+	}
+}
+
+//checkAppRunning wait for info of ZInfoApp type with mention of deployed AppName and ZSwState_RUNNING state
+func checkAppRunning(appName string) projects.ProcInfoFunc {
+	return func(msg *info.ZInfoMsg) error {
+		if msg.Ztype == info.ZInfoTypes_ZiApp {
+			if msg.GetAinfo().AppName == appName {
+				if msg.GetAinfo().State == info.ZSwState_RUNNING {
+					return fmt.Errorf("app RUNNING with name %s", appName)
+				}
+			}
+		}
+		return nil
+	}
+}
+
+//getEVEIP wait for IPs of EVE and returns them
+func getEVEIP(edgeNode *device.Ctx) projects.ProcTimerFunc {
+	return func() error {
+		if eveIPs := tc.GetState(edgeNode).GetEVEIPs(); len(eveIPs) > 0 {
+			externalIP = eveIPs[0]
+			return fmt.Errorf("external ip is: %s", externalIP)
+		}
+		return nil
+	}
+}
+
+//checkAppAccess try to access APP with timer
+func checkAppAccess() projects.ProcTimerFunc {
+	return func() error {
+		if externalIP == "" {
+			return nil
+		}
+		res, err := utils.RequestHTTPWithTimeout(fmt.Sprintf("http://%s:%d", externalIP, externalPort), time.Second)
+		if err != nil {
+			return nil
+		} else {
+			return fmt.Errorf(res)
+		}
+	}
+}
+
+//checkAppAbsent check if APP undefined in EVE
+func checkAppAbsent(appName string) projects.ProcInfoFunc {
+	return func(msg *info.ZInfoMsg) error {
+		if msg.Ztype == info.ZInfoTypes_ZiDevice {
+			for _, app := range msg.GetDinfo().AppInstances {
+				if app.Name == appName {
+					return nil
+				}
+			}
+			return fmt.Errorf("no app with %s found", appName)
+		}
+		return nil
+	}
+}
+
+//TestDockerStart gets EdgeNode and deploys app, defined in appLink
+//it generates random appName and adds processing functions
+//it checks if app processed by EVE, app in RUNNING state, app is accessible by HTTP get
+//it uses timewait for processing all events
+func TestDockerStart(t *testing.T) {
+	edgeNode := tc.GetEdgeNode(tc.WithTest(t))
+
+	tc.ConfigSync(edgeNode)
```

Comment:
Remind me, why do we need ConfigSync right after GetEdgeNode? Presumably there's no config (yet) to be synced, right?

Reply:
You are right, removed.